### PR TITLE
Switched the space shooter demo from none fixed process to fixed process.

### PR DESCRIPTION
**Pull request by jmintb**
The space shooter demo was stuttering alot, so I have minimized the
stuttering by switching from a none fixed process to a fixed process.

### CODE REVIEW

**Comment by neikeq** [comment]
Apparently the commit(s) are not being attributed to your Github account. The reason may be that your git configuration uses a different e-mail address to your Github account. [Read this article](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/) for more information.

However, that does not prevent from merging the Pull Request.

File: demos/2d/space_shooter/engine.cfg [hunk 1]
```diff
@@ -10,6 +10,10 @@ game_state="res://game_state.gd"
 
 [display]
 
+driver="GLES2"
+fullscreen=false
+text_edit_idle_detect_sec=3
+stretch_aspect="ignore"
```
Comment (neikeq) [inline demos/2d/space_shooter/engine.cfg]: Those seem to be the default values.

Comment (jmintb) [inline demos/2d/space_shooter/engine.cfg]: Is that a bad thing?

Comment (neikeq) [inline demos/2d/space_shooter/engine.cfg]: Not bad, but unnecessary since the values are the defaults :P

Comment (jmintb) [inline demos/2d/space_shooter/engine.cfg]: I havn't changed those so I am not sure why they are shown as have being changed.

**Comment by jmintb** [comment]
I think I have fixed my git config now, will that attribute the commits to my account retroactively?

**Comment by vnen** [comment]
> I think I have fixed my git config now, will that attribute the commits to my account retroactively?

You have to amend the commit. This should do the trick: `git commit --amend --reset-author`. Then you force push to your fork which updates this PR.

**Comment by jmintb** [comment]
Okay I dont mind if only my future commits are attributed to my github account. I am a bit confused about what just happend, I (think) I pulled down updates from the main godot repo to my fork, but then Travis started testing my commit again and it looks like I made a new commit. Does anyone know what happend? If this is too much off topic let me know and I will stop with the github related questions I was just curious. :)

**Comment by neikeq** [comment]
Try `git reset HEAD~` to remove the weird commit and then force push (with `-f`). Don't worry about asking git questions :)

**Comment by jmintb** [comment]
It looks like the commit and pull request have dissapeard after I force pushed. Should I make a new pull request?

**Comment by vnen** [comment]
@jmintb it seems you deleted the commit from your fork. If you add another commit to your master branch this PR can be reopened.

**Comment by jmintb** [comment]
I have made a commit to my master branch, but this PR is not opening. Other then making a new PR I am not sure what else to do?

**Comment by vnen** [comment]
So I reopened this but it has a bunch of conflicts because your fork is lagging behind and you pushed the changes in this repo to your fork in the same commit. Here's what you need to do:

```sh
git reset --soft HEAD^ # revert your commit without changing the files
git reset # clear staged files
git add demos/2d/space_shooter # stage your intended changes
git commit # add a commit with only your changes
git push --force # overwrite your GH fork with the new commit
```

After you do those steps this PR will update once again with only your changes which should not conflict with anything.

**Comment by jmintb** [comment]
WOW thanks for all the assistance, it looks like everything is in order :D.

**Comment by akien-mga** [comment]
Nice git crash course :D

File: demos/2d/space_shooter/engine.cfg (hunk 1 above)
Comment (akien-mga) [inline demos/2d/space_shooter/engine.cfg]: Hm, it's weird that they got added, normally the config only shows custom values.

File: demos/2d/space_shooter/level.tscn [hunk 2]
```diff
@@ -29,6 +30,27 @@ tile_data = IntArray( 16, 0, 17, 0, 18, 0, 19, 0, 20, 0, 21, 0, 22, 0, 23, 0, 24
 
 [node name="rail" parent="." instance=ExtResource( 2 )]
 
+[node name="ship" parent="rail"]
```
Comment (akien-mga) [inline demos/2d/space_shooter/level.tscn]: Did you make the changes below consciously, or did they happen automatically upon save?

Comment (jmintb) [inline demos/2d/space_shooter/level.tscn]: They must of happened automatically upon save.

Comment (akien-mga) [inline demos/2d/space_shooter/level.tscn]: Ok, it appears to be because you made the children of the `rail` instanced node modifiable in the level scene.

**Comment by akien-mga** [comment]
I cherry-picked the change in a9d246e to remove the unnecessary .tscn changes.

**Pull request closed without merging**